mono/mini/mini.h

```diff
@@ -3016,4 +3016,69 @@ mono_arch_load_function (MonoJitICallId jit_icall_id);
 MonoGenericContext
 mono_get_generic_context_from_stack_frame (MonoJitInfo *ji, gpointer generic_info);
 
+#if defined(__APPLE__) && defined(__arm64__)
+
+typedef enum {
+	JPM_NONE,
+	JPM_ENABLED,
+	JPM_DISABLED,
+} jit_protect_mode;
+
+extern __thread jit_protect_mode arm_current_jit_protect_mode;
+
+static void mono_arm_jit_write_protect_enable()
```

Comment:
nit - indentation.  more substantially we like `static inline` in headers

```suggestion
static inline void
mono_arm_jit_write_protect_enable()
```